squidDrone/src/i2c/i2c.cpp

```diff
@@ -0,0 +1,95 @@
+#include "i2c.hpp"
+
+namespace i2c
+{
+
+  auto I2C::Read(uint8_t address, uint16_t byte_size, uint32_t timeout) noexcept -> std::tuple<I2CStatus, std::vector<uint8_t>> {
+    I2CStatus i2c_status;
+
+    if (!CheckForValidInputRead(address, byte_size, timeout)) {
+      std::vector<uint8_t> data;
+      i2c_status = I2CStatus::I2C_PARAMETER_ERROR;
+      return {i2c_status, data};
+    }
+
+    address = ModifyAddressForI2C7Bit(address);
+
+    uint8_t buffer[byte_size];
+    HAL_StatusTypeDef hal_status = HAL_I2C_Master_Receive(&hi2c2, address, buffer, byte_size, timeout);
+
+    std::vector<uint8_t> data(buffer, buffer+byte_size);
+    i2c_status = GetI2CStatus(hal_status);
+
+    return {i2c_status, data};
+  }
+
+  auto I2C::Write(uint8_t address, const std::vector<uint8_t>& data, uint32_t timeout) noexcept -> I2CStatus {
+    if (!CheckForValidInputWrite(address, data, timeout)) {
+      return I2CStatus::I2C_PARAMETER_ERROR;
+    }
+
+    address = ModifyAddressForI2C7Bit(address);
+
+    HAL_StatusTypeDef hal_status = HAL_I2C_Master_Transmit(&hi2c2, address, (uint8_t*)data.data(), (uint16_t)data.size(), timeout);
+
+    return GetI2CStatus(hal_status);;
+  }
+
+  auto I2C::ModifyAddressForI2C7Bit(uint8_t address) noexcept -> uint8_t {
+    //because of 7 Bit addresses in I2C, one shift to the left
+    return (uint8_t)(address << 1);
```

Comment:
NO C-Style cast, use either static_cast or reinterpret_cast (static_cast is preferred if it is applicable)

Reply:
Fixed in this commit: https://github.com/elheck/Squiddrone/pull/219/commits/4443678ab2d3e710c132053c42a4729b33edd37c